File: help/pt_pt/exercicios_da_ficha_11/Ficha11_018/Ficha11_018.java
// Programador           -

import java.util.Scanner;

public class Ficha11_018 {

    static Scanner keyb = new Scanner(System.in);//keyboard

    //Main Function
    public static void main( String[] args) {
        int alt = 0 ;
        while(true ){
            System.out.print("Altura   : " );
            alt = Integer.valueOf(keyb.nextLine());
            if(alt % 2 == 0 && alt >= 4 ){
                break;
            }
            System.out.print("\tValor errado\n" );
        }
        System.out.print("Caractér : " );
        String txt = keyb.nextLine();
        copa(alt / 2 , txt ) ;
        tronco(alt / 2 , txt ) ;
    }

    //::::::::::::::::::::::::::::::::::::::::::::::::::
    //User Defined Functions Code
    //::::::::::::::::::::::::::::::::::::::::::::::::::
    public static void linhaCopa( int sp , int chars , String txt){
        for( int i = 0; i < sp; i = i + 1 ) {
            System.out.print(" " );
        }
        for( int i = 0; i < chars; i = i + 1 ) {
            System.out.print(txt );
        }
        System.out.print("\n" );
    }
    //::::::::::::::::::::::::::::::::::::::::::::::::::
    public static void copa( int alt , String txt){
        for( int i = 1; i < alt + 1; i = i + 1 ) {
            int sp = alt - i ;
            int chrs1 = i * 2 - 1 ;
            linhaCopa(sp , chrs1 , txt ) ;
        }
    }
    //::::::::::::::::::::::::::::::::::::::::::::::::::
    public static void tronco( int n , String ch){
        for( int i = 0; i < n; i = i + 1 ) {
            for( int j = 0; j < n - 1; j = j + 1 ) {
                System.out.print(" " );
            }
            System.out.print(ch + "\n" );
        }
    }

}
//Java  - Automatic translated by Algorithmi 22.05
//(c) Ant0nio M@nso 	quarta, 2022 out 12 20:17

File: help/pt_pt/exercicios_da_ficha_11/Ficha11_018/Ficha11_018.php
<?php
// Programador           -

//Main Function
function main() {
    $alt = 0 ;
    while(true ){
        echo  "Altura   : " ;
        $alt = intval(isset($_POST["alt"])? $_POST["alt"] : isset($_GET["alt"]) ? $_GET["alt"] : 0);
        
        echo $alt . "<br/>";// print input 
        if($alt % 2 == 0 && $alt >= 4 ){
            break;
        }
        echo  "    Valor errado\n" ;
    }
    echo  "Caractér : " ;
    $txt = isset($_POST["$txt"])? $_POST["$txt"] : isset($_GET["$txt"]) ? $_GET["$txt"] : 0;
    echo $txt . "<br/>";// print input 
    copa($alt / 2 , $txt ) ;
    tronco($alt / 2 , $txt ) ;
}

//::::::::::::::::::::::::::::::::::::::::::::::::::
//User Defined Functions Code
//::::::::::::::::::::::::::::::::::::::::::::::::::
function linhaCopa($sp, $chars, $txt) {
    for( $i = 0; $i < $sp; $i = $i + 1 ) {
        echo  " " ;
    }
    for( $i = 0; $i < $chars; $i = $i + 1 ) {
        echo $txt  ;
    }
    echo  "\n" ;
}
//::::::::::::::::::::::::::::::::::::::::::::::::::
function copa($alt, $txt) {
    for( $i = 1; $i < $alt + 1; $i = $i + 1 ) {
        $sp = $alt - $i ;
        $chrs1 = $i * 2 - 1 ;
        linhaCopa($sp , $chrs1 , $txt ) ;
    }
}
//::::::::::::::::::::::::::::::::::::::::::::::::::
function tronco($n, $ch) {
    for( $i = 0; $i < $n; $i = $i + 1 ) {
        for( j = 0; j < n - 1; j = j + 1 ) {
            echo  " " ;
        }
        echo $ch  .  "\n" ;
    }
}

main();

//PHP 5.3 +  - Automatic translated by Algorithmi 22.05
//(c) Luís Lopes 	quarta, 2022 out 12 20:17

?>